drivers/sensor/am2301b/am2301b.c

```diff
@@ -0,0 +1,188 @@
+/* am2301b.c - Driver for Aosong AM2301B Humidity and Temperature Module */
+/*
+ * Copyright (c) 2022 Hisanori Yamaoku
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aosong_am2301b
+
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#ifdef CONFIG_AM2301B_CRC_CHECK
+#include <sys/crc.h>
+#endif /* CONFIG_AM2301B_CRC_CHECK */
+
+#include "am2301b.h"
+
+LOG_MODULE_REGISTER(AM2301B, CONFIG_SENSOR_LOG_LEVEL);
+
+#define AM2301B_LEN_MEASUREMENT_DATA              7U
+
+const static uint8_t am2301b_cmd_initialize[] = { 0xBE, 0x08, 0x00 };
+const static uint8_t am2301b_cmd_measurement[] = { 0xAC, 0x33, 0x00 };
+
+static int am2301b_fetch_measument_data(const struct am2301b_config *cfg,
+					struct am2301b_data *data)
+{
+	uint8_t buf[AM2301B_LEN_MEASUREMENT_DATA];
+
+	if (i2c_write_dt(&cfg->i2c, am2301b_cmd_measurement, sizeof(am2301b_cmd_measurement))) {
+		LOG_ERR("Failed to send the command during measurement.");
+		return -EIO;
+	}
+
+	/* From Chapter 5.4, Section 3 of the AM2301B Product manual, */
+	/* wait for 80ms after sending the measurement command.       */
+	k_sleep(K_MSEC(AM2301B_WAIT_MEASUREMENT));
+
+	if (i2c_read_dt(&cfg->i2c, buf, sizeof(buf))) {
+		LOG_ERR("Failed to receive the measurement result.");
+		return -EIO;
+	}
+
+#ifdef CONFIG_AM2301B_CRC_CHECK
+	/* Calculate CRC from Chapter 5.4, Section 4 of AM2301B Product manuals. */
+	if (crc8(buf, 6, 0x31, 0xFF, false) != buf[6]) {
+		LOG_ERR("CRC verification failed.");
+		return -EIO;
+	}
+#endif /* CONFIG_AM2301B_CRC_CHECK */
+
+	if (buf[0] & AM2301B_MASK_MEASUREMENT_BUSY) {
+		LOG_ERR("A timeout was detected during measurement.");
+		return -EBUSY;
+	}
+
+	data->humidity = ((uint32_t)buf[1] << 12) | ((uint32_t)buf[2] << 4) | (buf[3] >> 4);
+	data->temperature = (uint32_t)(buf[3] & 0x0f) << 16 | ((uint32_t)buf[4] << 8) | buf[5];
```

Comment:
please use `sys/byteorder.h` APIs instead

Reply:
fixed it.